src/zpiv/zpivmodule.cpp

```diff
@@ -74,16 +74,19 @@ namespace ZPIVModule {
         return true;
     }
 
-    bool parseCoinSpend(const CTxIn &in, const CTransaction &tx, const CTxOut &prevOut, PublicCoinSpend &publicCoinSpend) {
-        if (!in.IsZerocoinPublicSpend() || !prevOut.IsZerocoinMint())
-            return error("%s: invalid argument/s\n", __func__);
-
+    PublicCoinSpend parseCoinSpend(const CTxIn &in){
         std::vector<char, zero_after_free_allocator<char> > data;
         data.insert(data.end(), in.scriptSig.begin() + 4, in.scriptSig.end());
         CDataStream serializedCoinSpend(data, SER_NETWORK, PROTOCOL_VERSION);
         libzerocoin::ZerocoinParams *params = Params().Zerocoin_Params(false);
-        PublicCoinSpend spend(params, serializedCoinSpend);
+        return PublicCoinSpend(params, serializedCoinSpend);
+    }
+
+    bool parseCoinSpend(const CTxIn &in, const CTransaction &tx, const CTxOut &prevOut, PublicCoinSpend &publicCoinSpend) {
+        if (!in.IsZerocoinPublicSpend() || !prevOut.IsZerocoinMint())
+            return error("%s: invalid argument/s\n", __func__);
```

Comment:
nit: No need for `\n` as that is already added in the `error()` function (same should be fixed in all other error messages in the file).

Reply:
done.